src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/CcdCaseUpdater.java

```diff
@@ -0,0 +1,239 @@
+                exception
+            );
+        }
+    }
+
+    private void updateCaseInCcd(
+        String service,
+        boolean ignoreWarnings,
+        String idamToken,
+        String s2sToken,
+        String userId,
+        ExceptionRecord exceptionRecord,
+        CaseUpdateDetails caseUpdateDetails,
+        StartEventResponse startEvent
+    ) {
+        CaseDetails existingCase = startEvent.getCaseDetails();
+
+        try {
+            coreCaseDataApi.submitForCaseworker(
+                idamToken,
+                s2sToken,
+                userId,
+                exceptionRecord.poBoxJurisdiction,
+                startEvent.getCaseDetails().getCaseTypeId(),
+                ignoreWarnings,
+                getCaseDataContent(exceptionRecord, caseUpdateDetails, startEvent)
+            );
+
+            log.info(
+                "Successfully updated case for service {} with case Id {} based on exception record ref {}",
+                service,
+                existingCase.getId(),
+                exceptionRecord.id
+            );
+        } catch (FeignException exception) {
+            log.error(
+                "Failed to update case for {} jurisdiction with case Id {} based on exception record with Id {}. "
+                    + "Service response: {}",
+                exceptionRecord.poBoxJurisdiction,
+                existingCase.getId(),
+                exceptionRecord.id,
+                exception.contentUTF8(),
+                exception
+            );
+
+            throw exception;
+        } catch (Exception exception) {
+            log.error(
+                "Failed to update case for {} jurisdiction with case Id {} based on exception record with Id {}",
+                exceptionRecord.poBoxJurisdiction,
+                existingCase.getId(),
+                exceptionRecord.id,
+                exception
+            );
+
+            throw exception;
+        }
+    }
+
+    private CaseDataContent getCaseDataContent(
+        ExceptionRecord exceptionRecord,
+        CaseUpdateDetails caseUpdateDetails,
+        StartEventResponse startEvent
+    ) {
+        return CaseDataContent
+            .builder()
+            .caseReference(exceptionRecord.id)
+            .data(caseUpdateDetails.caseData)
+            .event(getEvent(exceptionRecord, startEvent.getCaseDetails().getId(), startEvent.getEventId()))
+            .eventToken(startEvent.getToken())
+            .build();
+    }
+
+    private Event getEvent(
+        ExceptionRecord exceptionRecord,
+        Long existingCaseId,
+        String eventId
+    ) {
+        return Event
+            .builder()
+            .id(eventId)
+            .summary(format("Case updated, case Id %s", existingCaseId))
+            .description(
+                format(
+                    "Case with case Id %s updated based on exception record ref %s",
+                    existingCaseId,
+                    exceptionRecord.id
+                )
```

Comment:
The event is going to be created in the context of "this" case - we'll see it in the history of the case. Therefore, we don't need to include the case ID. Exception record ID is fine in the message.

Reply:
Will do